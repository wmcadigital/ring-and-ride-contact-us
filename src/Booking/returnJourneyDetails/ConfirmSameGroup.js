import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState, useForm } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import ProgressIndicator from "../../common/ProgressIndicator";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutReturnJourneySectionName from "./getAboutReturnJourneySectionName";

import { required } from "../../common/validation";

const ConfirmSameGroup = ({ setGroupSameAsOutward }) => {
  const stateApi = useFormState();
  const formApi = useForm();

  const formValues = stateApi.values;

  const error = stateApi.submitFailed
    ? stateApi.errors?.groupSameAsOutward
    : null;

  useEffect(() => {
    if (formValues.groupSameAsOutward) {
      if (!formValues.returnPassengers) {
        formApi.mutators.setFormAttribute(
          "returnPassengers",
          formValues.passengers
        );
      }
      setGroupSameAsOutward(formValues.groupSameAsOutward);
    }
  }, [formValues.groupSameAsOutward]);

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("3", stateApi)}
        sectionName={getAboutReturnJourneySectionName(stateApi)}
      />
      <Question text="Are the same number of passengers on the return journey?" />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Yes"
          validation={required}
          value="yes"
          fieldName="groupSameAsOutward"
        />
        <RadioButton
          key={2}
          label="No"
          validation={required}
          value="no"
          fieldName="groupSameAsOutward"
        />
      </RadioGroup>
    </FormSection>
  );
};

ConfirmSameGroup.propTypes = {
  setGroupSameAsOutward: PropTypes.func,
};

ConfirmSameGroup.defaultProps = {
  setGroupSameAsOutward: () => {},
};

export default ConfirmSameGroup;
