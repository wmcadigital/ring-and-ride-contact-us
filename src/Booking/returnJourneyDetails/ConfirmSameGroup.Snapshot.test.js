import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import ConfirmSameGroup from "./ConfirmSameGroup";

jest.mock("react-final-form");

describe("ConfirmSameGroup - snapshots", () => {
  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<ConfirmSameGroup />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
      },
      submitFailed: true,
      errors: { groupSameAsOutward: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<ConfirmSameGroup />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
