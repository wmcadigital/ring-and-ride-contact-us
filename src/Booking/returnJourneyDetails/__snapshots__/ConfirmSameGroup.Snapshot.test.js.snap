// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmSameGroup - snapshots individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 4 of 5"
  />
  <Question
    text="Are the same number of passengers on the return journey?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="groupSameAsOutward"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="groupSameAsOutward"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;

exports[`ConfirmSameGroup - snapshots renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their return journey"
    sectionPosition="Section 4 of 5"
  />
  <Question
    text="Are the same number of passengers on the return journey?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="groupSameAsOutward"
      label="Yes"
      onClickCallback={[Function]}
      validation={[Function]}
      value="yes"
    />
    <RadioButton
      fieldName="groupSameAsOutward"
      label="No"
      onClickCallback={[Function]}
      validation={[Function]}
      value="no"
    />
  </RadioGroup>
</FormSection>
`;
